x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -98,77 +141,169 @@ export const signalRulesAlertType = ({
+            tags,
+          });
+
+          if (bulkIndexResult) {
+            logger.debug(
+              `Finished signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            );
+            const sDate = new Date().toISOString();
+            currentStatusSavedObject.attributes.status = 'succeeded';
+            currentStatusSavedObject.attributes.statusDate = sDate;
+            currentStatusSavedObject.attributes.lastSuccessAt = sDate;
+            currentStatusSavedObject.attributes.lastSuccessMessage = 'succeeded';
+            await services.savedObjectsClient.update(
+              ruleStatusSavedObjectType,
+              currentStatusSavedObject.id,
+              {
+                ...currentStatusSavedObject.attributes,
+              }
+            );
+          } else {
+            logger.error(
+              `Error processing signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            );
+            const sDate = new Date().toISOString();
+            currentStatusSavedObject.attributes.status = 'failed';
+            currentStatusSavedObject.attributes.statusDate = sDate;
+            currentStatusSavedObject.attributes.lastFailureAt = sDate;
+            currentStatusSavedObject.attributes.lastFailureMessage = 'There was an error!!';
+            // current status is failing
+            await services.savedObjectsClient.update(
+              ruleStatusSavedObjectType,
+              currentStatusSavedObject.id,
+              {
+                ...currentStatusSavedObject.attributes,
+              }
+            );
+            // create new status for historical purposes
+            await services.savedObjectsClient.create(ruleStatusSavedObjectType, {
+              ...currentStatusSavedObject.attributes,
+            });
+
+            if (ruleStatusSavedObjects.saved_objects.length >= 6) {
+              // delete fifth status and prepare to insert a newer one.
+              const toDelete = ruleStatusSavedObjects.saved_objects.slice(5);
+              await toDelete.forEach(async item =>
+                services.savedObjectsClient.delete(ruleStatusSavedObjectType, item.id)
+              );
+            }
+          }
+        } catch (err) {
+          // TODO: Error handling and writing of errors into a signal that has error
+          // handling/conditions
           logger.error(
-            `Error processing signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+            `Error from signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
+          );
+          const sDate = new Date().toISOString();
+          currentStatusSavedObject.attributes.status = 'failed';
+          currentStatusSavedObject.attributes.statusDate = sDate;
+          currentStatusSavedObject.attributes.lastFailureAt = sDate;
+          currentStatusSavedObject.attributes.lastFailureMessage = JSON.stringify(err, null, 4);
```

Comment:
JavaScript trivia bug, paste this into your browser:

```ts
JSON.stringify(new Error('Hello how are you?'))
"{}"
```

doing a stringify on an error never does what you would hope it would do. You want to change this to a simple text error message for now but not stringify `new Error` types as they won't do anything.